x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -324,6 +328,61 @@ const analyzeDegradedFieldRoute = createDatasetQualityServerRoute({
   },
 });
 
+const updateFieldLimitRoute = createDatasetQualityServerRoute({
+  endpoint: 'PUT /internal/dataset_quality/data_streams/{dataStream}/update_field_limit',
+  params: t.type({
+    path: t.type({
+      dataStream: t.string,
+    }),
+    body: t.type({
+      newFieldLimit: t.number,
+      indexTemplate: t.string,
+      lastBackingIndex: t.string,
```

Comment:
I understand your concern. But firstly this API is an Internal API which means clients should not call this on their own and secondly gathering this information again on the server side means additional API calls to be done which is expensive. 

The UI does not let the user touch any of the other parameters except the field limit value itself

Reply:
> this API is an Internal API which means clients should not call this on their own.

It doesn't prevent them to call it anyway if they want from inside Kibana. Just because they shouldn't do it, it doesn't imply they can't. 

> gathering this information again on the server side means additional API calls to be done which is expensive.

I'm not strongly concerned about the performance here, we are applying changes on the cluster and is expected it might take some time (which should not, as retrieving the settings and index details can be parallelized and shouldn't take long). Also, the user is given feedback that something is going on with the loading indicator, so the additional time is a matter of perception for the user. The update limit change is also something that ideally shouldn't happen often, so there is no risk to overload the cluster with many requests.

Here is a video of what is possible if we leave the API open:

https://github.com/user-attachments/assets/8f07a510-d375-4560-9c68-685d68e221fc

I believe we should make impossible for an API to break certain changes we apply at the cluster level from the client app.

Reply:
To be honest, the users if they chose to play with internal APIs which they are not supposed to should also understand the risks no ?

I understand that this can be done via Dev Tools and so are direct PUT and POST requests.

How different is it for a user to do the same thing via the Dev Tools using a Elasticsearch PUT or POST request

```
PUT _component_template/log-whatever@custom
{
  "template": {
    "settings": {
      "index.mapping.total_fields.limit": 2000
    }
  }
}
```

Reply:
They can surely break a cluster configuration manually if they want. What concerns me is the de-synced API we leave on the client. 
If one day we update something on the state management of the client app and for some reason the React state that set these values is not aligned on `indexTemplate` and `lastBackingIndex` we would introduce the above bug we have in the recordings and that would be kind of breaking. 

I believe this should be a fairly quick refactor and it would bring more stability to the API, but happy to discuss it more with the team if you want.

Reply:
Yes, the state machine mess up is the concern which i also thought about over the weekend. It can easily happen that the UI state goes out if sync. I will move this to the server side